dns/provisionroute53_policy.tf

```diff
@@ -0,0 +1,22 @@
+# ------------------------------------------------------------------------------
+# Create the IAM policy that allows all of the Route 53 actions necessary to
+# provision Delegation Sets in the DNS account
+# ------------------------------------------------------------------------------
+
+data "aws_iam_policy_document" "provisionroute53_doc" {
+  statement {
+    actions = [
+      "route53:CreateReusableDelegationSet",
+      "route53:GetReusableDelegationSet",
+      "route53:ListReusableDelegationSets"
+    ]
```

Comment:
I think it is worth adding a comment explaining why you chose not to include `route53:DeleteReusableDelegationSet` here.